install.py

```
@@ -43,47 +46,88 @@ def evaluate(self, npm_available: bool) -> tuple[bool, str | None]:
+    return target_dir / "npm" / "package" / "bin" / "npm-cli.js"
+
+
+def extract_tarball_safely(fileobj: io.BytesIO, destination: Path) -> None:
+    """Extract a tarball while rejecting path traversal entries."""
+    destination = destination.resolve()
+    with tarfile.open(fileobj=fileobj, mode="r:gz") as tar:
+        for member in tar.getmembers():
+            member_path = (destination / member.name).resolve()
+            if not member_path.is_relative_to(destination):
+                raise ValueError(f"Unsafe tar entry: {member.name}")
+        tar.extractall(destination)
```

Ruff (0.15.5)

[error] Uses of `tarfile.extractall()`

(S202)